Docs/Book/C++Examples/example12.cpp

```diff
@@ -0,0 +1,56 @@
+//
+// Preserving molecules - example12.cpp
+
+#include <fstream>
+#include <iostream>
+#include <string>
+#include <GraphMol/GraphMol.h>
+#include <GraphMol/MolPickler.h>
+#include <GraphMol/FileParsers/MolSupplier.h>
+#include <GraphMol/SmilesParse/SmilesParse.h>
+#include <GraphMol/SmilesParse/SmilesWrite.h>
+
+int main( int argc , char **argv ) {
+
+  RDKit::ROMOL_SPTR mol1( RDKit::SmilesToMol( "c1ccncc1" ) );
+  std::string pickle;
+  RDKit::MolPickler::pickleMol( *mol1 , pickle );
+  RDKit::ROMol mol2;
+  RDKit::MolPickler::molFromPickle( pickle , mol2 );
+  std::cout << RDKit::MolToSmiles( mol2 ) << std::endl;
+
+  // writing to pickle file
+  std::string smi_file = getenv("RDBASE");
+  smi_file += "/Code/GraphMol/test_data/canonSmiles.long.smi";
+  std::string pkl_name = "canonSmiles.long.pkl";
```

Comment:
It would be better to not name this `.pkl` since that could mislead someone coming from the python side into thinking that they'd be able to read the file easily (they can't). `.bin` would be better.